### PR TITLE
[WFCORE-5508] Add missing static final

### PR DESCRIPTION
**Pull request by boris-unckel**
Fixes https://issues.redhat.com/browse/WFCORE-5508

### CODE REVIEW

**Comment by boris-unckel** [comment]
@jmesnil Hello Jeff, tests passed for this PR, failures seem not related. Could you review and approve, please?

File: cli/src/main/java/org/jboss/as/cli/gui/JConsoleCLIPlugin.java [hunk 1]
```diff
@@ -74,7 +74,7 @@
 
     // Global count of created pools
     private static final AtomicInteger executorCount = new AtomicInteger();
-    private final Logger log = Logger.getLogger(JConsoleCLIPlugin.class.getName());
+    private static final Logger log = Logger.getLogger(JConsoleCLIPlugin.class.getName());
```
Comment (jmesnil) [inline cli/src/main/java/org/jboss/as/cli/gui/JConsoleCLIPlugin.java:77]: I'm not sure this is required: the `log` field is used as an instance field. We do not need to make it static.

If we do, let's rename it to `LOG` as this is our convention for constants.

File: cli/src/main/java/org/jboss/as/cli/handlers/LsHandler.java [hunk 2]
```diff
@@ -63,7 +63,7 @@
     private final ArgumentWithValue nodePath;
     private final ArgumentWithoutValue l;
     private final ArgumentWithoutValue resolve;
-    private final Logger log = Logger.getLogger(LsHandler.class);
+    private static final Logger log = Logger.getLogger(LsHandler.class);
```
Comment (jmesnil) [inline cli/src/main/java/org/jboss/as/cli/handlers/LsHandler.java:66]: idem

File: cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java [hunk 3]
```diff
@@ -290,7 +290,7 @@
     /** operation request address prefix */
     private final OperationRequestAddress prefix = new DefaultOperationRequestAddress();
     /** the prefix formatter */
-    private final NodePathFormatter prefixFormatter = DefaultPrefixFormatter.INSTANCE;
+    private static final NodePathFormatter prefixFormatter = DefaultPrefixFormatter.INSTANCE;
```
Comment (jmesnil) [inline cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java:293]: idem

File: cli/src/main/java/org/jboss/as/cli/impl/DeploymentItemCompleter.java [hunk 4]
```diff
@@ -42,7 +42,7 @@
 public class DeploymentItemCompleter implements CommandLineCompleter {
 
     private final OperationRequestAddress address;
-    private final Logger log = Logger.getLogger(DeploymentItemCompleter.class);
+    private static final Logger log = Logger.getLogger(DeploymentItemCompleter.class);
```
Comment (jmesnil) [inline cli/src/main/java/org/jboss/as/cli/impl/DeploymentItemCompleter.java:45]: idem

File: core-model-test/framework/src/main/java/org/jboss/as/core/model/test/TestModelControllerService.java [hunk 5]
```diff
@@ -408,7 +408,7 @@ private void delete(File file) {
     private class ServerInitializer implements Initializer {
         final ExtensibleConfigurationPersister persister = new NullConfigurationPersister();
         final ServerEnvironment environment = createStandaloneServerEnvironment();
-        final boolean parallelBoot = false;
+        static final boolean parallelBoot = false;
```
Comment (jmesnil) [inline core-model-test/framework/src/main/java/org/jboss/as/core/model/test/TestModelControllerService.java:411]: I'm not sure we need to flag it as static.

File: core-model-test/framework/src/main/java/org/jboss/as/core/model/test/TestModelControllerService.java [hunk 6]
```diff
@@ -443,7 +443,7 @@ public void initCoreModel(Resource rootResource, ManagementResourceRegistration
     }
 
     private class HostInitializer implements Initializer {
-        final String hostName = "master";
+        static final String hostName = "master";
```
Comment (jmesnil) [inline core-model-test/framework/src/main/java/org/jboss/as/core/model/test/TestModelControllerService.java:446]: idem

File: domain-management/src/main/java/org/jboss/as/domain/management/security/UserLdapCallbackHandler.java [hunk 7]
```diff
@@ -88,7 +88,7 @@
 
     private final boolean allowEmptyPassword;
     private final boolean shareConnection;
-    protected final int searchTimeLimit = 10000; // TODO - Maybe make configurable.
+    protected static final int searchTimeLimit = 10000; // TODO - Maybe make configurable.
```
Comment (jmesnil) [inline domain-management/src/main/java/org/jboss/as/domain/management/security/UserLdapCallbackHandler.java:91]: as far as I can tell, this field is not used at all. I'm not sure we need to make it static as it's not clear what its usage would be.

@darranl could you shed some light on it? Could we delete it completely?

File: host-controller/src/main/java/org/jboss/as/host/controller/RemoteDomainConnection.java [hunk 8]
```diff
@@ -481,7 +481,7 @@ public void execute(ManagementRequestContext<Void> voidManagementRequestContext)
      private class CompleteRegistrationRequest extends AbstractManagementRequest<Void, Void> {
 
          private final byte outcome;
-         private final String message = "yay!"; //
+         private static final String message = "yay!"; //
```
Comment (jmesnil) [inline host-controller/src/main/java/org/jboss/as/host/controller/RemoteDomainConnection.java:484]: static is not needed

File: protocol/src/main/java/org/jboss/as/protocol/mgmt/ManagementClientChannelStrategy.java [hunk 9]
```diff
@@ -139,7 +139,7 @@ private static ProtocolConnectionConfiguration createConfiguration(final Protoco
      */
     private static class Establishing extends FutureManagementChannel {
 
-        private final String serviceType = DEFAULT_CHANNEL_SERVICE_TYPE;
+        private static final String serviceType = DEFAULT_CHANNEL_SERVICE_TYPE;
```
Comment (jmesnil) [inline protocol/src/main/java/org/jboss/as/protocol/mgmt/ManagementClientChannelStrategy.java:142]: static is not needed

File: server/src/main/java/org/jboss/as/server/mgmt/domain/HostControllerConnection.java [hunk 10]
```diff
@@ -327,7 +327,7 @@ public void execute(ManagementRequestContext<Void> voidManagementRequestContext)
 
     public class ServerStartedRequest extends AbstractManagementRequest<Void, Void> {
 
-        private final String message = ""; // started / failed message
+        private static final String message = ""; // started / failed message
```
Comment (jmesnil) [inline server/src/main/java/org/jboss/as/server/mgmt/domain/HostControllerConnection.java:330]: static is not needed

**Review by jmesnil — CHANGES_REQUESTED** [review]

File: cli/src/main/java/org/jboss/as/cli/gui/JConsoleCLIPlugin.java (hunk 1 above)
Comment (boris-unckel) [reply cli/src/main/java/org/jboss/as/cli/gui/JConsoleCLIPlugin.java:77]: @jmesnil Thanks for your fast feedback. The pattern behind the missing static (both for LOG and the Strings below) is a final together with a direct assignment is indicating a constant. And constants shall be reused per class not per instance.
What is your thought when a immutable Object is assigned final to keep it as instance? Maybe I'm missing something.

File: domain-management/src/main/java/org/jboss/as/domain/management/security/UserLdapCallbackHandler.java (hunk 7 above)
Comment (darranl) [reply domain-management/src/main/java/org/jboss/as/domain/management/security/UserLdapCallbackHandler.java:91]: I am not going to be looking at this, at the moment my tasks are to remove legacy security.

**Comment by darranl** [comment]
Some of these statics do seem more problematic than they are worth, in some cases it appears now references will be retained as long as the class is loaded instead for as long as there is an instance on the heap.

For these kinds of "clean up" it may be better to have a discussion on the wildfly-dev list - if there are bugs in the code fixes are great but when we are cleaning up based on third party coding conventions and recommendations it is probably worth checking how far this needs to go.

All of this code being changed is in the versions we need to maintain for support purposes with code being ported between these branches now at risk of merge conflicts - if we are not addressing a bug we should be sure of the benefit.

**Comment by boris-unckel** [comment]
@jmesnil @darranl Thanks for feedback and suggestions how to deal with cases like this one in the future, I will follow that. I understand that cost/risk/benefit ratio is not enough to go forward with this PR. I withdraw it.

**Pull request closed without merging**